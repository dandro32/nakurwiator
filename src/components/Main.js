import React from 'react';
import ROUTES from '../configs/routes.config';
import { Route } from 'react-router-dom';

const Main = () => {
  const routes = ROUTES.map((route, i) => {
    return(
      <Route
        key={i}
        path={route.path}
        render={props => (
          <route.component {...props} />
        )}
      />
    );
  });
  return (
    <main>
      {routes}
    </main>
  );
};

export default Main;