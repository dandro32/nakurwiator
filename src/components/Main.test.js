import React from 'react';
import Main from './Main';
import ShallowRenderer from 'react-test-renderer/shallow';

describe('Main component tests', ()=> {
  let shallowRendererResult;
  beforeEach(() => {
    const shallowRenderer = new ShallowRenderer();
    shallowRenderer.render(<Main />);
    shallowRendererResult = shallowRenderer.getRenderOutput();
  });

  it('renders correctly', () => {
    expect(shallowRendererResult).toMatchSnapshot();
  });
});