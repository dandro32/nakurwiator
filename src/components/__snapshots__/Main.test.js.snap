// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main component tests renders correctly 1`] = `
<main>
  <Redirect
    from="/"
    push={false}
    to="/trainings"
  />
  <Route
    path="/trainings"
    render={[Function]}
  />
  <Route
    path="/exercises"
    render={[Function]}
  />
  <Route
    path="/raports"
    render={[Function]}
  />
  <Route
    path="/extras"
    render={[Function]}
  />
  <Route
    path="/settings"
    render={[Function]}
  />
</main>
`;
